### PR TITLE
[12.0][MIG] sale_order_line_price_history

### PR DESCRIPTION
**Pull request by murtuzasaleh**
Sale order line price history
=====================

### CODE REVIEW

File: sale_order_line_price_history/__manifest__.py [hunk 1]
```diff
@@ -2,14 +2,15 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 {
     "name": "Sale order line price history",
-    "version": "11.0.1.0.0",
+    "version": "12.0.1.0.0",
     "category": "Sales Management",
     "author": "Tecnativa, "
               "Odoo Community Association (OCA)",
     "website": "https://github.com/OCA/sale-workflow/",
+    "summary": "Sale order line price history",
```
Comment (pedrobaeza) [inline sale_order_line_price_history/__manifest__.py]: Don't repeat the same as the name. It's not required and I prefer to not put it.

File: sale_order_line_price_history/__manifest__.py [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
     "license": "AGPL-3",
     "depends": [
-        "sale",
+        "sale_management",
```
Comment (pedrobaeza) [inline sale_order_line_price_history/__manifest__.py]: Why is this needed?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: sale_order_line_price_history/__manifest__.py (hunk 1 above)
Comment (murtuzasaleh) [reply sale_order_line_price_history/__manifest__.py]: @pedrobaeza Ok, I will remove it.

File: sale_order_line_price_history/__manifest__.py (hunk 2 above)
Comment (murtuzasaleh) [reply sale_order_line_price_history/__manifest__.py]: @pedrobaeza To display the Main Sales menu.

Comment (Swapnesh-SerpentCS) [reply sale_order_line_price_history/__manifest__.py]: @murtuzasaleh  Have a look at https://github.com/OCA/sale-workflow/pull/606#issuecomment-364018576

File: sale_order_line_price_history/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Sale order line price history",
+    "version": "12.0.1.0.0",
+    "category": "Sales Management",
+    "author": "Tecnativa, "
```
Comment (Swapnesh-SerpentCS) [inline sale_order_line_price_history/__manifest__.py]: Blank space at end of the line after `Tecnativa,`

File: sale_order_line_price_history/wizards/__init__.py [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from . import sale_orderLine_price_history
```
Comment (Swapnesh-SerpentCS) [inline sale_order_line_price_history/wizards/__init__.py]:
```suggestion
from . import sale_order_line_price_history
```

I mean to rename the file and change it here

File: sale_order_line_price_history/wizards/sale_orderLine_price_history.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        return self.env['sale.order.line'].browse(line_id).order_partner_id.id
+
+    sale_order_line_id = fields.Many2one(
+        comodel_name='sale.order.line',
+        string='Sale order line',
+        default=lambda self: self.env.context.get("active_id"),
+    )
+    product_id = fields.Many2one(
+        related="sale_order_line_id.product_id",
+        readonly=True,
```
Comment (Swapnesh-SerpentCS) [inline sale_order_line_price_history/wizards/sale_orderLine_price_history.py]: Related fields are read only by default

File: sale_order_line_price_history/wizards/sale_orderLine_price_history.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Customer',
+        default=_default_partner_id,
+    )
+    line_ids = fields.One2many(
+        comodel_name="sale.order.line.price.history.line",
+        inverse_name="history_id",
+        string="History line",
```
Comment (Swapnesh-SerpentCS) [inline sale_order_line_price_history/wizards/sale_orderLine_price_history.py]:
```suggestion
        string="History lines",
```

**Review by Swapnesh-SerpentCS — COMMENTED** [review]
Minor Improvements

**Comment by murtuzasaleh** [comment]
> Minor Improvements

Done.

**Review by Swapnesh-SerpentCS — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]
